arch/arm/src/armv7-a/arm_cache.c

```diff
@@ -190,8 +198,16 @@ void up_clean_dcache_all(void)
 
 void up_flush_dcache(uintptr_t start, uintptr_t end)
 {
-  cp15_flush_dcache(start, end);
-  l2cc_flush(start, end);
+  if (cp15_cache_size() < (end - end))
```

Comment:
```suggestion
  if (cp15_cache_size() < (end - start))
```

Reply:
I think that test cases have a big gap if this was not catched during testing.

Reply:
I think if the buffer being flushed (end - start)  larger than cache size,  the entire cache is used by the buffer.